src/app/core/cart/cart.service.ts

```diff
@@ -61,12 +66,16 @@ export class CartService {
 
     private evaluateCartWeight(items: any): number {
         return items
-            .filter((item: any) => item?.dimensions?.weight != null && item?.quantity != null)
-            .reduce((sum: number, item: any) => sum + (item.dimensions.weight * item.quantity), 0);
+            .map((item: any) => item.quantity * this.getSnipcartItemWeight(item))
+            .reduce((sum: number, weight: number) => sum + weight, 0);
+    }
+
+    private getSnipcartItemWeight(item: any): number {
+        let weightField = item.customFields.find((field: any) => field.name == "Weight");
```

Comment:
weightField is never reassigned so we should use const here.

Also, except when checking for null/undefined, we should always use strict equality: field.name === "Weight".